__tests__/components/ResourceCreation/PatientCreation.test.tsx

```diff
@@ -152,4 +176,74 @@ describe('PatientCreation', () => {
       expect(download).toBeCalledTimes(1);
     });
   });
+
+  it('should have calculate function called when calcuate button is clicked', async () => {
+    const MockPatients = getMockRecoilState(patientTestCaseState, {
+      'example-pt': {
+        patient: {
+          resourceType: 'Patient',
+          name: [{ given: ['Test123'], family: 'Patient456' }]
+        },
+        resources: [
+          {
+            resourceType: 'Procedure',
+            id: 'test-id',
+            status: 'completed',
+            subject: {}
+          }
+        ]
+      }
+    });
+
+    const MockSelectedPatient = getMockRecoilState(selectedPatientState, 'example-pt');
+
+    const MockMB = getMockRecoilState(measureBundleState, {
+      name: 'testName',
+      content: testBundle as fhir4.Bundle
+    });
+
+    // Mock calculate data requirements because of the changing state
+    jest.spyOn(Calculator, 'calculateDataRequirements').mockResolvedValue({
+      results: {
+        resourceType: 'Library',
+        status: 'draft',
+        type: {}
+      }
+    });
+
+    jest.spyOn(Calculator, 'calculateMeasureReports').mockResolvedValue({
+      results: [MOCK_MEASURE_REPORT]
+    });
+
+    await act(async () => {
+      render(
+        mantineRecoilWrap(
+          <>
+            <MockPatients />
+            <MockSelectedPatient />
+            <MockMB />
+            <MeasureUpload />
+            <PatientCreation {...DEFAULT_PROPS} isPatientModalOpen={false} currentPatient={null} />
+          </>
+        )
+      );
+    });
+
+    const calculateButton = screen.getByTestId('calculate-button') as HTMLButtonElement;
```

Comment:
In general we shouldn't default to using a `testid` where possible. One of the main paradigms of react unit testing is to test things the user sees (for example, a button that says "Calculate"). A user can't see a testid, and so it's use should really be for when we don't have another way to access this element within a unit test. Try this instead, does it still work?

``` typescript
const calculateButton = screen.getByRole('button', { name: 'Calculate' }) as HTMLButtonElement
```

does it still work?